File: Interview Preparation Kit/Arrays/ArrayManipulation.py
#!/bin/python3
# https://www.hackerrank.com/challenges/crush/problem
import math
import os
import random
import re
import sys

# Complete the arrayManipulation function below.
def arrayManipulation(n, queries):
    largest = 0
    array = [0] * n

    queries_len = len(queries)
    for i in range(queries_len):
        a = queries[i][0]
        b = queries[i][1]
        k = queries[i][2]

        array[a - 1] += k
        if (b < n):
            array[b] -= k

    curr_largest = 0
    for j in range(n):
        curr_largest += array[j]
        largest = max(curr_largest, largest)

    return largest


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    nm = input().split()

    n = int(nm[0])

    m = int(nm[1])

    queries = []

    for _ in range(m):
        queries.append(list(map(int, input().rstrip().split())))

    result = arrayManipulation(n, queries)

    fptr.write(str(result) + '\n')

    fptr.close()

File: Interview Preparation Kit/Arrays/ArrayManipulation.java
// https://www.hackerrank.com/challenges/crush/problem
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;

public class Solution {

    // Complete the arrayManipulation function below.
    static long arrayManipulation(int n, int[][] queries) {
        long largest = 0;
        long[] array = new long[n];

        for (int i = 0; i < queries.length; i++) {
            int a = queries[i][0];
            int b = queries[i][1];
            long k = (long) queries[i][2];

            array[a - 1] += k;
            if (b < n) {
                array[b] -= k;
            }
        }

        long curr_largest = 0;
        for (int j = 0; j < n; j++) {
            curr_largest += array[j];
            largest = Math.max(curr_largest, largest);
        }

        return largest;
    }

    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) throws IOException {
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        String[] nm = scanner.nextLine().split(" ");

        int n = Integer.parseInt(nm[0]);

        int m = Integer.parseInt(nm[1]);

        int[][] queries = new int[m][3];

        for (int i = 0; i < m; i++) {
            String[] queriesRowItems = scanner.nextLine().split(" ");
            scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");

            for (int j = 0; j < 3; j++) {
                int queriesItem = Integer.parseInt(queriesRowItems[j]);
                queries[i][j] = queriesItem;
            }
        }

        long result = arrayManipulation(n, queries);

        bufferedWriter.write(String.valueOf(result));
        bufferedWriter.newLine();

        bufferedWriter.close();

        scanner.close();
    }
}
